service-base/src/main/java/org/eclipse/hono/service/plan/PrometheusBasedResourceLimitChecks.java

```diff
@@ -0,0 +1,173 @@
+            .replace(".", "_");
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    private final WebClient client;
+    private String host = "prometheus-operated.hono";
+    private int port = 9090;
+
+    /**
+     * Creates a new PrometheusBasedResourceLimitChecks instance.
+     *
+     * @param vertx The Vertx instance to use.
+     * @throws NullPointerException if any of the parameters are {@code null}.
+     */
+    public PrometheusBasedResourceLimitChecks(final Vertx vertx) {
+        this.client = WebClient.create(Objects.requireNonNull(vertx));
+    }
+
+    @Override
+    public Future<?> isConnectionLimitExceeded(final TenantObject tenant) {
+        Objects.requireNonNull(tenant);
+        log.trace("Connections limit for tenant [{}] is [{}]", tenant.getTenantId(), tenant.getConnectionsLimit());
+        if (tenant.getConnectionsLimit() != -1) {
+            return queryForSumInPrometheus(CONNECTIONS_METRIC_NAME, tenant)
+                    .recover(failure -> Future.succeededFuture(0L))
+                    .compose(noOfConnections -> {
+                        if (Optional.ofNullable(noOfConnections).orElse(0L) < tenant.getConnectionsLimit()) {
+                            return Future.succeededFuture();
+                        } else {
+                            log.trace(
+                                    "Connections limit exceeded for tenant. [tenant: {}, no.-Connections:{}, connections-limit:{}]",
+                                    tenant.getTenantId(), noOfConnections, tenant.getConnectionsLimit());
+                            return Future
+                                    .failedFuture(new ClientErrorException(HttpURLConnection.HTTP_FORBIDDEN,
+                                            String.format("Connections limit exceeded for tenant: [%s]",
+                                                    tenant.getTenantId())));
+                        }
+                    });
+        }
+        return Future.succeededFuture();
+    }
+
+    /**
+     * Gets the host of the prometheus backend.
+     *
+     * @return host The host of the Prometheus backend.
+     */
+    public String getHost() {
+        return host;
+    }
+
+    /**
+     * Sets the host of the prometheus backend.
+     *
+     * @param host The host of the Prometheus backend.
+     */
+    public void setHost(final String host) {
+        this.host = Objects.requireNonNull(host);
+    }
+
+    /**
+     * Gets the port of the prometheus backend.
+     *
+     * @return port The port of the Prometheus backend.
+     */
+    public int getPort() {
+        return port;
+    }
+
+    /**
+     * Sets the port of the prometheus backend.
+     *
+     * @param port The port of the Prometheus backend.
+     * @throws IllegalArgumentException if the port number is &lt; 0 or &gt; 2^16 - 1
+     */
+    public void setPort(final int port) {
```

Comment:
please document the default value (`9090`)

Reply:
@sophokles73 The default values are documented and I pushed the changes.